### DISCUSSION TITLE
How to add custom LR scheduler?

### DISCUSSION THREAD

**Discussion by JasspalSingh** [facebookresearch/detectron2#3490] [Q&A]
I have configured the code for custom Lr scheduler 

> def build_lr_scheduler(cls, cfg, optimizer):
>         """
>         Build a LR scheduler from config.
>         """
>         name = cfg.SOLVER.LR_SCHEDULER_NAME
>         if name == "ReduceLROnPlateau":
>             return ReduceLROnPlateau(
>                 optimizer,
>                 mode='min', factor=cfg.SOLVER.GAMMA, patience=1000, 
>                 threshold=0.001, threshold_mode='rel', cooldown=0, 
>                 min_lr=1e-06, eps=1e-08,verbose=True
>             )
>         else:
>             raise ValueError("Unknown LR scheduler: {}".format(name))
> 

 

- [ ] how can I employ this to update my learning rate.

I am getting error and solution to it is we need to give loss to 'ReduceLROnPlateau' at every iteration. How can I do it?
